src/LocalDatastoreController.default.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) 2015-present, Parse, LLC.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+const memMap = {};
+const LocalDatastoreController = {
```

Comment:
what should be the point of this local datastore? I'm not sure it has any used as on any page refresh it would be empty.

Reply:
There are some instances when localStorage isn't available. 

When I was making this localStorage was always undefined unless I mocked it.

I wanted to have some sort of backup. Might be used on in cloud code on a server somewhere.

Reply:
Right, that kinda sucks... https://developer.mozilla.org/en-US/docs/Web/API/Storage/LocalStorage

on the web it's been available since forever, so you're safe. 

so:

- browser: we're safe
- node: we probably want to use this in memory one or an LRU.
- react-native: we want async storage.

Because we want a different module based on the platform, we should use compilation flags to determine the proper one (through the usage of process.env.PARSE_BUILD) (like in the rest controller when it picks the right XMLHTTPRequest implementation).